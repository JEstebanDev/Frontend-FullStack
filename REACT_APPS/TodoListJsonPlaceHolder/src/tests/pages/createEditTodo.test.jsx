import {
  findAllByText,
  fireEvent,
  render,
  screen,
} from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CreateEditTodo from "../../pages/createEditTodo";
import userEvent from "@testing-library/user-event";
import { expect } from "vitest";
import { vi } from "vitest";
describe("testing the component", () => {
  test("should render the CreateEditTodo and take snapshot", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/create"]}>
        <CreateEditTodo />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test("should show error message when the id is empty", () => {
    render(
      <MemoryRouter initialEntries={["/create"]}>
        <CreateEditTodo />
      </MemoryRouter>
    );

    const idValue = screen.getByPlaceholderText(/id user/i);
    fireEvent.change(idValue, { target: { value: "" } });
    idValue.querySelector("#sandbox > form > span:nth-child(2)");
  });

  test("should display correct error message", async () => {
    const { getByTestId, findAllByText } = render(
      <MemoryRouter initialEntries={["/create"]}>
        <CreateEditTodo />
      </MemoryRouter>
    );
    fireEvent.click(getByTestId("submit"));
    await findAllByText("This field is required");
  });

  test("should call the submit button", () => {
    const mockFn = vi.fn();

    render(
      <MemoryRouter initialEntries={["/create"]}>
        <CreateEditTodo />
      </MemoryRouter>
    );

    const form = screen.getByRole("form");
    userEvent.submit(form);
    expect(mockFn).toHaveBeenCalled();
  });
});
