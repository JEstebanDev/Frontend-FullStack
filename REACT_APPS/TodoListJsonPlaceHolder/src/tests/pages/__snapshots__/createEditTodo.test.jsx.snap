// Vitest Snapshot v1

exports[`testing the component > should render the CreateEditTodo and take snapshot 1`] = `
<div>
  <div
    class="card info-form animate__animated animate__bounceIn"
  >
    <h3>
      Create
       a todo task 
      <i
        class="ri-add-box-fill"
      />
    </h3>
    <form
      aria-label="form"
    >
      <input
        class="inputs"
        data-testid="idtask"
        name="id"
        placeholder="id user"
      />
      <textarea
        class="inputs"
        data-testid="textodo"
        name="task"
        placeholder="write your task"
        type="text"
      />
      <br />
      <select
        class="inputs"
        name="completed"
      >
        <option
          value="true"
        >
          Yes
        </option>
        <option
          value="false"
        >
          No
        </option>
      </select>
      <br />
      <br />
      <input
        class="button editButton"
        data-testid="submit"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
